internal/runners/exec/exec.go

```diff
@@ -123,7 +124,8 @@ func (s *Exec) Run(params *Params, args ...string) error {
 		if err != nil {
 			return locale.WrapError(err, "err_initialize_runtime_event_handler")
 		}
-		if err := rt.Update(s.auth, eh); err != nil {
+		sockPath := svcctl.NewIPCSockPathFromGlobals().String()
+		if err := rt.Update(s.auth, eh, sockPath); err != nil {
```

Comment:
Is there a use-case for passing this as an argument rather than having the runtime call `svcctl.NewIPCSockPathFromGlobals().String()` itself?

Reply:
This is to avoid having the svcctl dependency expressed so deep in the stack.

Reply:
That's a cost without a value in my book. We can always go that route if we have a use-case for it, it wouldn't require much refactoring. Now we're having to do this every time we init a runtime, which adds specialized knowledge. Take BE for example developing the offline installer, they'd need to understand the basics of our IPC socket just to use runtimes.

Please do this in the runtime itself.